pkg/prometheus/promcfg.go

```diff
@@ -4092,3 +4044,29 @@ func getLowerByteSize(v *monitoringv1.ByteSize, cpf *monitoringv1.CommonPromethe
 func isByteSizeEmpty(v *monitoringv1.ByteSize) bool {
 	return v == nil || *v == ""
 }
+
+func (cg *ConfigGenerator) addFiltersToYaml(cfg yaml.MapSlice, filters []monitoringv1alpha1.Filter) yaml.MapSlice {
+	if len(filters) == 0 {
+		return cfg
+	}
+
+	// Sort the filters by name to generate deterministic config.
+	sort.Slice(filters, func(i, j int) bool {
+		return filters[i].Name < filters[j].Name
+	})
+
+	filtersYamlMap := []yaml.MapSlice{}
+	for _, filter := range filters {
+		filtersYamlMap = append(filtersYamlMap, yaml.MapSlice{
+			{
+				Key:   "name",
+				Value: filter.Name,
+			},
+			{
+				Key:   "values",
+				Value: filter.Values,
+			}})
+	}
```

Comment:
let's avoid the extra step
```suggestion
	filtersYamlMap := []yaml.MapSlice{}
	for _, filter := range filters {
		filtersYamlMap = append(filtersYamlMap, yaml.MapSlice{
			{
				Key:   "name",
				Value: filter.Name,
			},
			{
				Key:   "values",
				Value: filter.Values,
			}})
	}
```

Reply:
AFAICT it's not required since it happens when serializing to YAML anyway

Reply:
I removed the sorting logic and generated the config and `a_dummy_label` comes after `dummy_label`. 
Seems like the filters are not sorted, or do you mean something else?

Reply:
My bad! Without the explicit sorting, the keys should always be listed as they are in the custom resource. While it's unlikely that the order changes repeatedly in the spec, it's always better to ensure predictability of the generated configuration.

suggestion: use https://pkg.go.dev/slices#SortStableFunc and https://pkg.go.dev/cmp#Compare